backend/src/main/java/ai/verta/modeldb/datasetVersion/DatasetVersionDAORdbImpl.java

```diff
@@ -122,12 +121,12 @@ public DatasetVersion createDatasetVersion(
           getDatasetVersionFromRequest(authService, request, userInfo, existingDatasetVersion);
 
       if (datasetVersionList.size() == 1) {
-        transaction.commit();
-        return datasetVersionList.get(0);
+        DatasetVersion datasetVersion = datasetVersionList.get(0);
+        createRoleBindingsForDatasetVersion(userInfo, datasetVersion);
```

Comment:
when datasetVersionList size is 1 it means the new version was similar to the old version, in which case we need not create the bindings as was happening in older code.

Reply:
Remove RoleBinding creation step